### PR TITLE
[SFTP] fix error when listing empty path

### PR DESCRIPTION
**Pull request by JbIPS**
When trying to list or stat with an empty path as param, it will apply it on the root path instead of throwing an error.

Closes https://github.com/silexlabs/Silex/issues/974

### CODE REVIEW

File: lib/unifile-sftp.js [hunk 1]
```diff
@@ -27,6 +27,10 @@ function parseError(err) {
 	throw error;
 }
 
+function protectPath(path) {
+	return path.length ? path : '/';
```
Comment (lexoyo) [inline lib/unifile-sftp.js]: Shouldn't it throw an error? Is it consistent with the other services?

**Comment by JbIPS** [comment]
Yes, if this fix is greenlit I'll create a new patch version

File: lib/unifile-sftp.js (hunk 1 above)
Comment (JbIPS) [reply lib/unifile-sftp.js]: It was already throwing an error. But we had this kind of behavior in [Dropox](https://github.com/silexlabs/unifile/blob/master/lib/unifile-dropbox.js#L150) or [GitHub](https://github.com/silexlabs/unifile/blob/master/lib/unifile-github.js#L25) (but in the other way)

**Comment by JbIPS** [comment]
> Would it be useful to do the same with ftp?

FTP is OK for me, with a local VSFTPD server and my OVH instance

**Pull request merged**